### PR TITLE
feat(#8447): accept nepali digits in medic id and lmp date in SMS messages

### PR DESCRIPTION
**Pull request by yrimal**
# Description

Allow nepali digits in medic id and lmp date

medic/cht-core#8447

# Code review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# Compose URLs

If Build CI hasn't passed, these may 404:

* __CHT_CORE_COMPOSE_URL__
* __COUCH_SINGLE_COMPOSE_URL__
* __COUCH_CLUSTER_COMPOSE_URL__

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

File: api/tests/mocha/services/report/smsparser.spec.js [hunk 1]
```diff
@@ -613,13 +613,49 @@ describe('sms parser', () => {
...
+
   it('parse BS date parts yyys 2: textforms', () => {
     const doc = { message: '#ID 12345 #Y 2068 #M 11 #D 29' };
     const def = definitions.forms.YYYS;
     const data = smsparser.parse(def, doc);
     chai.expect(data).to.deep.equal({
-      patient_id: 12345,
-      lmp_year: 2068, lmp_month: 11, lmp_day: 29,
+      patient_id: '12345',
+      lmp_year: '2068', lmp_month: '11', lmp_day: '29',
```
Comment (garethbowen) [inline api/tests/mocha/services/report/smsparser.spec.js:658]: This change is concerning to me - is there any way that the standardiseDigits function could be called with a number in production code, or was this an inaccuracy in the test code?

**Review by garethbowen — CHANGES_REQUESTED** [review]
Nice PR, thanks!

Do we also need to change the `bsDate` function in the same way?

File: api/tests/mocha/services/report/smsparser.spec.js (hunk 1 above)
Comment (yrimal) [reply api/tests/mocha/services/report/smsparser.spec.js:658]: I checked how these fields are stored ultimately with current change and before, and they're all stored in quotation in production in couchdb, so I guess this was inaccuracy in test code.

**Review by dianabarsan — COMMENTED** [review]
Great work! Just one small request for a unit test inline.

File: api/tests/mocha/services/report/smsparser.spec.js [hunk 2]
```diff
@@ -613,13 +613,49 @@ describe('sms parser', () => {
...
+  });
+
   it('parse BS date parts yyys 2: textforms', () => {
     const doc = { message: '#ID 12345 #Y 2068 #M 11 #D 29' };
     const def = definitions.forms.YYYS;
     const data = smsparser.parse(def, doc);
     chai.expect(data).to.deep.equal({
-      patient_id: 12345,
-      lmp_year: 2068, lmp_month: 11, lmp_day: 29,
+      patient_id: '12345',
```
[the remark this answers is not in the document]
Comment (yrimal) [reply api/tests/mocha/services/report/smsparser.spec.js:677]: Thanks for feedback, I've added unittests for bsDate as well @dianabarsan .

**Review by garethbowen — APPROVED** [review]
Nice! Thanks.

**Review by dianabarsan — APPROVED** [review]
Very nice!

**Pull request merged**